### PR TITLE
[SPARK-46546][DOCS] Fix the formatting of tables in `running-on-yarn` pages

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to fix the formatting of tables in `running-on-yarn` pages.

### Why are the changes needed?
Make the tables on the page display normally.
Before:
image

After:
image

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Manually check.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
1.https://spark.apache.org/docs/latest/running-on-kubernetes.html#configuration
Before:
image

After:
image

2.https://spark.apache.org/docs/latest/running-on-yarn.html#spark-properties
Before:
image

After:
image

**Comment by panbingkun** [comment]
cc @zhengruifeng @HyukjinKwon @dongjoon-hyun @LuciferYang

**Comment by nchammas** [comment]
I think a better approach to fixing the styling of tables is to assign them to a class and then define the appropriate rules in [`custom.css`][1]. It will be much simpler and more robust than manually massaging the contents of each cell.

[1]: https://github.com/apache/spark/blob/af3a22533ca0a11d91cc70c920d5423a8dac9ee5/docs/css/custom.css

I was planning to do this as a follow-up to #44300 if it gets approved.

**Comment by bjornjorgensen** [comment]
There are actually more places where this issue are. Like in [GROUP BY Clause for spark 3.5](https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-groupby.html)
where is looks like this
image

**Comment by panbingkun** [comment]
I will define a style in the file `custom.css` and try it out.

**Comment by yaooqinn** [comment]
Hi @panbingkun, the scrollbar has been fixed. FYI,https://github.com/apache/spark/pull/44423

**Comment by panbingkun** [comment]
> Hi @panbingkun, the scrollbar has been fixed. FYI,#44423

Okay, but I generated a document with the latest code a few days ago, and the table format seems to be incorrect.

**Comment by panbingkun** [comment]
> > Hi @panbingkun, the scrollbar has been fixed. FYI,#44423
> 
> @yaooqinn Okay, but I generated a document with the latest code a few days ago, and the table format seems to be incorrect.

@yaooqinn
Indeed, I just checked again and the master's logic is now scrollable.
But I found another issue, I will reuse this PR and modify the following issue:

image

**Comment by yaooqinn** [comment]
@panbingkun, thank you for the double check

File: docs/running-on-yarn.md [hunk 1]
```diff
@@ -875,6 +875,7 @@ The following extra configuration options are available when the shuffle service
     initialization. This prevents application failures caused by running containers on
     NodeManagers where the Spark Shuffle Service is not running.
   </td>
+  <td></td>
```
Comment (yaooqinn) [inline docs/running-on-yarn.md:878]: Please ensure that the value is fulfilled.

Comment (LuciferYang) [reply docs/running-on-yarn.md:878]: +1

Comment (panbingkun) [reply docs/running-on-yarn.md:878]: Okay, let me find the versions of each option.

File: docs/running-on-yarn.md [hunk 2]
```diff
@@ -866,7 +866,7 @@ to avoid garbage collection issues during shuffle.
 The following extra configuration options are available when the shuffle service is running on YARN:
 
 <table>
-<thead><tr><th>Property Name</th><th>Default</th><th>Meaning</th></tr></thead>
+<thead><tr><th>Property Name</th><th>Default</th><th>Meaning</th><th>Since Version</th></tr></thead>
 <tr>
   <td><code>spark.yarn.shuffle.stopOnFailure</code></td>
```
Comment (panbingkun) [inline docs/running-on-yarn.md:871]: For a record: https://github.com/apache/spark/pull/14162

File: docs/running-on-yarn.md [hunk 3]
```diff
@@ -875,6 +875,7 @@ The following extra configuration options are available when the shuffle service
     initialization. This prevents application failures caused by running containers on
     NodeManagers where the Spark Shuffle Service is not running.
   </td>
+  <td></td>
 </tr>
 <tr>
   <td><code>spark.yarn.shuffle.service.metrics.namespace</code></td>
```
Comment (panbingkun) [inline docs/running-on-yarn.md:881]: https://github.com/apache/spark/pull/31936

File: docs/running-on-yarn.md [hunk 4]
```diff
@@ -883,6 +884,7 @@ The following extra configuration options are available when the shuffle service
     The namespace to use when emitting shuffle service metrics into Hadoop metrics2 system of the
     NodeManager.
   </td>
+  <td></td>
 </tr>
 <tr>
   <td><code>spark.yarn.shuffle.service.logs.namespace</code></td>
```
Comment (panbingkun) [inline docs/running-on-yarn.md:890]: https://github.com/apache/spark/pull/34079/files#diff-661e1c80840458affe09c28d71a7901d9bf15d6e87edd0c26dec1c8fd78b128aR810

File: docs/running-on-yarn.md (hunk 1 above)
Comment (panbingkun) [reply docs/running-on-yarn.md]: Done

**Review by yaooqinn — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master/3.5.

**Comment by dongjoon-hyun** [comment]
Thank you, @panbingkun and all.

**Pull request closed without merging**